### ISSUE TITLE
Building from source: exe symlinks don't honor DESTDIR

### ISSUE THREAD

**Issue by kierdavis** [aws/aws-cli#8467]
### Describe the bug

In regard to building & installing the v2 CLI from source, the [recommended behaviour of the `DESTDIR` option][makedoc] isn't honored when creating the `$prefix/bin/aws -> $prefix/lib/aws-cli/bin/aws` symlink, and likewise for `aws_completer`.

[makedoc]: https://www.gnu.org/software/make/manual/html_node/DESTDIR.html#DESTDIR

### Expected Behavior

The symlink created at `$DESTDIR/$prefix/bin/aws` should point to `$prefix/lib/aws-cli/bin/aws`.

More generally, the contents of `$DESTDIR` should not themselves contain any references to `$DESTDIR`, so that it's possible to copy the contents of this directory to another machine's root filesystem (typically via a RPM/DEB/etc) and still end up with a working installation.

### Current Behavior

It points to `$DESTDIR/$prefix/lib/aws-cli/bin/aws`.

If `$DESTDIR` is packaged up and installed onto another machine, `$prefix/bin/aws` will be a broken symlink.

### Reproduction Steps

```
$ git checkout 2.15.6
$ ./configure --prefix=/usr/local --with-download-deps
...
$ make
...
$ make DESTDIR=$PWD/install_staging install
...
$ readlink ./install_staging/usr/local/bin/aws
/home/kdavis/checkouts/github/aws/aws-cli/install_staging/usr/local/lib/aws-cli/bin/aws
```

### Possible Solution

Use relative symlinks.

### CLI version used

2.15.6

### Environment details (OS name and version, etc.)

Linux - RHEL 9

**Comment by tuananh** [comment]
I also faced this issue when trying to build from src.

**Comment by amberkushwaha** [comment]
The $prefix/bin/aws -> $prefix/lib/aws-cli/bin/aws symlink (and likewise for aws_completer)

The script interpreter of $prefix/lib/aws-cli/bin/aws (and likewise for aws_completer)
the script is being prepared in the same obligations and marks as used before formats.

Expected Behavior
The symlink created at $DESTDIR/$prefix/bin/aws should point to $prefix/lib/aws-cli/bin/aws.

The first line of $DESTDIR/$prefix/lib/aws-cli/bin/aws should be #!$prefix/lib/aws-cli/bin/python.

More generally, the contents of $DESTDIR should not themselves contain any references to $DESTDIR, so that it's possible to copy the contents of this directory to another machine's root filesystem (typically via a RPM/DEB/etc) and still end up with a working installation.

This code of conduct is still the in the initial stage so please do the needful for the coding ground formats and tempers.

**Comment by Stealthii** [comment]
For anyone looking towards a workaround for this issue until AWS fix it, you only need to redirect the symlinks and fix the shebangs to address the reference issue.
```bash
# Building package
build_dir="$(mktemp -d)"
target_dir="$(mktemp -d)"
tarball_uri="https://github.com/aws/aws-cli/archive/refs/tags/${awscli_version}.tar.gz"
# Download and extract tarball
curl -sL "${tarball_uri}" | tar -C "${build_dir}" -xz
# Configure and make install to target directory
(
        set -e
        cd "${build_dir}/aws-cli-${awscli_version}"
        export PYTHON="${python}"
        ./configure --prefix "/usr" --with-download-deps
        make
        make install DESTDIR="${target_dir}"
)

# Fix broken DESTDIR behaviour in aws-cli
for f in $(find "${target_dir}" -type l); do
        # Fix symlinks
        ln -sfn "$(readlink -f "${f}" | sed "s#${target_dir}##")" "${f}"
done
for f in $(grep -rl "${target_dir}" ${target_dir}); do
        # Fix shebangs
        sed -i "s|${target_dir}||g" $f
done
```

**Comment by tim-finnigan** [comment]
Thanks for reporting this issue and your patience here, will plan to bring this up for discussion with the team.

**Comment by Stealthii** [comment]
See [GNU's DESTDIR documentation](https://www.gnu.org/prep/standards/html_node/DESTDIR.html) for reference on this coding standard.

The expectation is that you can define an installation prefix, such as `/usr`:
```sh
./configure --prefix "/usr"
```
But run the `make install` against a DESTDIR as a buildroot target for packaging (such as `/tmp/awscli`):
```sh
make install DESTDIR="/tmp/awscli"
```

---

With regards to re-producing the issue, [the above comment](#issuecomment-2017961269) sets up an example build & target install using this standard pattern.

The issue stems from parts of your build process adhering to DESTDIR (for example the `make install` correctly places files under the DESTDIR tree), however your Make (or the build_system) does not adhere to the prefix when setting up symlinks.

**TL;DR**: `DESTDIR` should be used to define a given installation destination directory, not to be confused with `prefix` which defines the target installation base path.

**Comment by ReakNar** [comment]
# Building package
build_dir="$(mktemp -d)"
target_dir="$(mktemp -d)"
tarball_uri="https://github.com/aws/aws-cli/archive/refs/tags/${awscli_version}.tar.gz"
# Download and extract tarball
curl -sL "${tarball_uri}" | tar -C "${build_dir}" -xz
# Configure and make install to target directory
(
        set -e
        cd "${build_dir}/aws-cli-${awscli_version}"
        export PYTHON="${python}"
        ./configure --prefix "/usr" --with-download-deps
        make
        make install DESTDIR="${target_dir}"
)

# Fix broken DESTDIR behaviour in aws-cli
for f in $(find "${target_dir}" -type l); do
        # Fix symlinks
        ln -sfn "$(readlink -f "${f}" | sed "s#${target_dir}##")" "${f}"
done
for f in $(grep -rl "${target_dir}" ${target_dir}); do
        # Fix shebangs
        sed -i "s|${target_dir}||g" $f
done
